### PR TITLE
Add documentation for building native on Windows

### PR DESCRIPTION
**Pull request by gdoenlen**
Now that compiling to native on windows is working properly add
documentation on the steps required as discussed in #5675

### CODE REVIEW

**Comment by geoand** [comment]
Thanks for the contribution!

Could someone with a Windows machine verify this? Perhaps @rsvoboda or @maxandersen ?

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 1]
```diff
@@ -47,6 +47,7 @@ sudo apt-get install build-essential libz-dev zlib1g-dev
 ----
 xcode-select --install
 ----
+* On Windows you'll need to install the https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=BuildTools&rel=16[Visual Studio Build Tools]
 ====
```
Comment (maxandersen) [inline docs/src/main/asciidoc/building-native-image.adoc]: I installed windows 7 sdk based on instructions given from graalvm team info - but that was months ago;  has it changed ?

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 2]
```diff
@@ -75,6 +76,13 @@ On macOS, point the variable to the `Home` sub-directory:
 ----
 export GRAALVM_HOME=$HOME/Development/graalvm/Contents/Home/
 ----
++
+On Windows, you'll have to go through the Control Panel to set your environment variables.
++
+[TIP]
+====
+Installing via scoop will do this for you.
```
Comment (maxandersen) [inline docs/src/main/asciidoc/building-native-image.adoc]: a google search didn't reveal to me how to install graalvm with scoop - got a pointer ?

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 3]
```diff
@@ -47,6 +47,7 @@ sudo apt-get install build-essential libz-dev zlib1g-dev
 ----
 xcode-select --install
 ----
+* On Windows you'll need to install the https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=BuildTools&rel=16[Visual Studio Build Tools]
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc]:
```suggestion
* On Windows, you will need to install the https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=BuildTools&rel=16[Visual Studio Build Tools]
```

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 4]
```diff
@@ -60,7 +61,7 @@ If you cannot install GraalVM, you can use a multi-stage Docker build to run Mav
 Version {graalvm-version} is required. Using the community edition is enough.
 
 1. Install GraalVM if you haven't already. You have a few options for this:
-** Use platform-specific install tools like https://github.com/graalvm/homebrew-tap[homebrew] or https://sdkman.io/jdks#Oracle[sdkman].
+** Use platform-specific install tools like https://github.com/graalvm/homebrew-tap[homebrew] or https://sdkman.io/jdks#Oracle[sdkman] or https://github.com/ScoopInstaller/Java[scoop].
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc]:
```suggestion
** Use platform-specific install tools like https://github.com/graalvm/homebrew-tap[homebrew], https://sdkman.io/jdks#Oracle[sdkman] or https://github.com/ScoopInstaller/Java[scoop].
```

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 5]
```diff
@@ -75,6 +76,13 @@ On macOS, point the variable to the `Home` sub-directory:
 ----
 export GRAALVM_HOME=$HOME/Development/graalvm/Contents/Home/
 ----
++
+On Windows, you'll have to go through the Control Panel to set your environment variables.
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc]:
```suggestion
On Windows, you will have to go through the Control Panel to set your environment variables.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks for taking the time to share this.

I added some very minor wording suggestions. Could you apply them and squash? Thanks!

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 2 above)
Comment (gdoenlen) [reply docs/src/main/asciidoc/building-native-image.adoc]: Above I linked to the scoop bucket for Java. Once you install scoop, you'll have to enable the bucket `scoop bucket add java` and then install graal: `scoop install graalvm`. That installs the latest release (jdk 11), but theres also `graalvm-nightly`, `graalvm8`, `graalvm8-nightly`, `graalvm11`, and `graalvm11-nightly`.

Comment (maxandersen) [reply docs/src/main/asciidoc/building-native-image.adoc]: very nice! good to know!

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 1 above)
Comment (gdoenlen) [reply docs/src/main/asciidoc/building-native-image.adoc]: No the official message from the SubstrateVM team has not changed according to the github readme, but according to this [thread](https://github.com/oracle/graal/issues/1850) it looks like the the Windows 7 sdk is only required for older versions of GraalVM and with Java 8. 

It looks like if you want to run JDK 11 on a modern graal you'll need at least VS 2017:

> The JDK 8 versions used Visual Studio 10. The Java team builds JDK 11 with Visual Studio 2017. There is a Windows implementation of GraalVM 19.3 that uses JDK 8. You can try this as a work-around until you get a newer VS version.

So I could update this here to link to VS 2017 and add instructions based on the JDK wanting to be used.

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 4 above)
Comment (rsvoboda) [inline docs/src/main/asciidoc/building-native-image.adoc]: For Windows I would mention Chocolatey tool, not just scoop. 
Chocolatey is very popular on Windows platform - 6.3k stars on Chocolatey GH vs. 358 stars for scoop.

https://chocolatey.org/packages/graalvm

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 6]
```diff
@@ -158,6 +166,20 @@ just pass -Dquarkus.package.type=native as a property on the command line, howev
...
+.Issues with packaging on Windows
+====
+The Microsoft Native Tools for Visual Studio must first be initialized before packaging. You can do this by starting
+the `x64 Native Tools Command Prompt` that was installed with the Visual Studio Build Tools.
+
+Another solution is to write a script to do this for you:
+
+[source,shell]
+----
+cmd /c '"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Auxiliary\Build\vcvars64.bat" && mvn package -Pnative'
```
Comment (rsvoboda) [inline docs/src/main/asciidoc/building-native-image.adoc]: For VS 2017 the path is `call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvars64.bat"`

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 4 above)
Comment (maxandersen) [reply docs/src/main/asciidoc/building-native-image.adoc]: scoop has the advantage to not require admin so of such its a simpler model.

Comment (gdoenlen) [reply docs/src/main/asciidoc/building-native-image.adoc]: The main scoop github repo actually has [9.9k](https://github.com/lukesampson/scoop) stars. It's also much easier to contribute to (in fact I just had the PR to export GRAALVM_HOME accepted last night), but I can include choco as well.

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 6 above)
Comment (gdoenlen) [reply docs/src/main/asciidoc/building-native-image.adoc]: Will do.

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 4 above)
Comment (rsvoboda) [reply docs/src/main/asciidoc/building-native-image.adoc]: ah, I looked at https://github.com/ScoopInstaller

not require admin - that's a strong argument for `scoop`

Consider my comment about `choco` inclusion as optional

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 1 above)
Comment (gdoenlen) [reply docs/src/main/asciidoc/building-native-image.adoc]: @gsmet How would you like this worded?

```
* On Windows you will need to install the Visual Studio Build Tools:
** For JDK 8 based projects you need to install the https://www.microsoft.com/en-us/download/details.aspx?id=8442[Windows 7 SDK]
** For JDK 11 based projects you need to install the https://aka.ms/vs/15/release/vs_buildtools.exe[Visual Studio 2017 Build Tools]
```

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 7]
```diff
@@ -158,6 +166,20 @@ just pass -Dquarkus.package.type=native as a property on the command line, howev
 this allows native image tests to also be run.
 
 Create a native executable using: `./mvnw package -Pnative`.
+[[graal-and-windows]]
+[NOTE]
+.Issues with packaging on Windows
+====
+The Microsoft Native Tools for Visual Studio must first be initialized before packaging. You can do this by starting
+the `x64 Native Tools Command Prompt` that was installed with the Visual Studio Build Tools.
```
Comment (gdoenlen) [inline docs/src/main/asciidoc/building-native-image.adoc]: @gsmet Wording here will be a little odd as well with the Windows 7 SDK vs Visual Studio 2017 Build Tools.

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 6 above)
Comment (gdoenlen) [reply docs/src/main/asciidoc/building-native-image.adoc]: @maxandersen Is there a different path for the Windows 7 vars file?

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 1 above)
Comment (gsmet) [reply docs/src/main/asciidoc/building-native-image.adoc]: I would just keep it simple and only describe what needs to be done for JDK 11 projects.

JDK 8 is deprecated in master so no need to make things more awkward than necessary.

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 6 above)
Comment (gdoenlen) [reply docs/src/main/asciidoc/building-native-image.adoc]: @rsvoboda I updated this to `cmd /c 'call "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build\vcvars64.bat" && mvn package -Pnative'` as that's where the file lives when only installing the build tools.

```ps1
C:\Program Files (x86)\Microsoft Visual Studio\2017> gci -r | where name -eq 'vcvars64.bat'


    Directory: C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---           3/18/2020  9:10 AM             39 vcvars64.bat
```

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
@gdoenlen merged, thanks a lot!

**Pull request merged**